### PR TITLE
Fix and untag spec for Enumerable#count unused block warning

### PR DESCRIPTION
**Pull request by havenwood**

### CODE REVIEW

File: src/main/ruby/core/enumerable.rb [hunk 1]
```diff
@@ -112,6 +112,7 @@ def count(item = undefined)
     seq = 0
     if !undefined.equal?(item)
       each { |o| seq += 1 if item == o }
+      warn 'warning: given block not used'
```
Comment (eregon) [inline src/main/ruby/core/enumerable.rb]: Should this check if a block was actually given?

Comment (havenwood) [inline src/main/ruby/core/enumerable.rb]: Yes it absolutely should.

File: src/main/ruby/core/enumerable.rb [hunk 2]
```diff
@@ -112,6 +112,7 @@ def count(item = undefined)
     seq = 0
     if !undefined.equal?(item)
       each { |o| seq += 1 if item == o }
+      warn 'warning: given block not used' if block_given?
```
Comment (eregon) [inline src/main/ruby/core/enumerable.rb]: I think it should also warn as soon as possible, and in particular before calling `each`.

Comment (havenwood) [inline src/main/ruby/core/enumerable.rb]: I tried to match the MRI behavior which only warns after iteration stops. I did expect it to warn up front.

```ruby
[sleep(1), sleep(1)].count(nil) { }
# after two seconds:
#!> warning: given block not used
```

Comment (havenwood) [inline src/main/ruby/core/enumerable.rb]: I wonder about fixing this upstream. Should I change it to use the intuitive behavior or do as CRuby does for now?

Comment (eregon) [inline src/main/ruby/core/enumerable.rb]: This is due to the Array literal being evaluated before the call to `count`.
The code below shows the warning is printed before calling each:
```ruby
[3] pry(main)> obj = Object.new
=> #<Object:0x00559bf4245a28>
[4] pry(main)> def obj.each; p :each; end
=> :each
[5] pry(main)> obj.extend Enumerable
=> #<Object:0x00559bf4245a28>
[6] pry(main)> obj.count(nil) { }
(pry):6: warning: given block not used
:each
```
or maybe more simply:
```ruby
[7] pry(main)> obj = Object.new
=> #<Object:0x00559bf3ad9230>
[8] pry(main)> obj.extend Enumerable
=> #<Object:0x00559bf3ad9230>
[9] pry(main)> obj.count(nil) { }
(pry):9: warning: given block not used
NoMethodError: undefined method `each' for #<Object:0x00559bf3ad9230>
```

**Pull request merged**